src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringExtractor.cs

```diff
@@ -0,0 +1,197 @@
+                // Include "name" under the root.
+                new StringFilteredTraversalRule("name"),
+                // Include "description" under the root.
+                new StringFilteredTraversalRule("description"),
+                // Include "symbols" under the root, if they also comply with child rules.
+                new StringFilteredTraversalRule("symbols").WithChild(
+                    // Any symbol is included, skip none.
+                    new AllInclusiveTraversalRule().WithChildren(
+                        // Include "displayName" of each symbol.
+                        new StringFilteredTraversalRule("displayName"),
+                        // Include "description" of each symbol.
+                        new StringFilteredTraversalRule("description"),
+                        // Include "choices" of symbols, if they also comply with child rules.
+                        new StringFilteredTraversalRule("choices", new MemberBasedKeyCreator("choice")).WithChild(
+                            // Include any element of the "choices" array. No choice will be skipped.
+                            new AllInclusiveTraversalRule().WithChildren(
+                                // Include "displayName" of each choice.
+                                new StringFilteredTraversalRule("displayName"),
+                                // Include "description" of each choice.
+                                new StringFilteredTraversalRule("description"))))),
+                // Include "postActions" under the root, if they also comply with child rules.
+                new StringFilteredTraversalRule("postActions").WithChild(
+                    // Any post action in "postActions" array should be included. Skip none.
+                    new AllInclusiveTraversalRule().WithChildren(
+                        // Include "description" of the post action.
+                        new StringFilteredTraversalRule("description"),
+                        // Include "manualInstructions" of the post action, if they also comply with child rules.
+                        new RegexFilteredTraversalRule("manualInstructions").WithChild(
+                            // Include all the manual instructions in the array. Skip none.
+                            new AllInclusiveTraversalRule().WithChild(
+                                // Include "text" of the post action.
+                                new StringFilteredTraversalRule("text"))))));
+
+        public TemplateStringExtractor(JsonDocument document, ILoggerFactory? loggerFactory = null)
+        {
+            _jsonDocument = document;
+            _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+            _logger = _loggerFactory.CreateLogger<TemplateStringExtractor>();
+        }
+
+        public IReadOnlyList<TemplateString> ExtractStrings()
+        {
+            List<TemplateString>? extractedStrings = new();
```

Comment:
Why is that question mark there? And can we stop using `new();` and instead use `var` please?

Reply:
Removed the question mark. Let's make a team decision to use or not use `new ()`. I will update the feature branch before it is merged to main accordingly.